### PR TITLE
Add internal/config/sidecar test

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added `internal/config/sidecar_test.go` and godoc comment.
**grammer check passed**

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/config/sidecar_test.go [hunk 1]
```diff
@@ -55,57 +57,165 @@ func TestAgentSidecar_Bind(t *testing.T) {
...
+		}(),
+		func() test {
+			mode := "sidecar"
+			watchDir := "sidecar"
+			autoBackupDuration := "10ms"
+			postStopTimeout := "5m"
+			filename := "vald-ngt-1"
+			filenameSuffix := "tar.gz"
+			return test{
+				name: "return AgentSidecar when all of object are set",
```
Comment (hlts2) [inline internal/config/sidecar_test.go:125]:
```suggestion
				name: "return AgentSidecar when all of object are not set",
```

File: internal/config/sidecar_test.go [hunk 2]
```diff
@@ -55,57 +57,165 @@ func TestAgentSidecar_Bind(t *testing.T) {
...
+						Filename:           filename,
+						FilenameSuffix:     filenameSuffix,
+						BlobStorage:        new(Blob),
+						Compress:           new(CompressCore),
+						RestoreBackoff:     new(Backoff),
+						Client:             new(Client),
+					},
+				},
+			}
+		}(),
```
Comment (vankichi) [inline internal/config/sidecar_test.go:149]: seems not much between test name and the fields parameter

File: internal/config/sidecar_test.go [hunk 3]
```diff
@@ -55,57 +57,165 @@ func TestAgentSidecar_Bind(t *testing.T) {
...
+		func() test {
+			mode := "sidecar"
+			watchDir := "sidecar"
+			autoBackupDuration := "10ms"
+			postStopTimeout := "5m"
+			filename := "vald-ngt-1"
+			filenameSuffix := "tar.gz"
+			return test{
+				name: "return AgentSidecar when all of object are not set",
```
Comment (hlts2) [inline internal/config/sidecar_test.go:125]:
```suggestion
				name: "return AgentSidecar when the BlobStorage and Compress and RestoreBackoff and Client are not set",
```

File: internal/config/sidecar_test.go (hunk 2 above)
Comment (hlts2) [reply internal/config/sidecar_test.go]: thank you. I fixed.

**Review by vankichi — APPROVED** [review]

**Pull request merged**